dali/pipeline/operators/reader/loader/sequence_loader.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_READER_LOADER_SEQUENCE_LOADER_H_
+#define DALI_PIPELINE_OPERATORS_READER_LOADER_SEQUENCE_LOADER_H_
+
+#include <numeric>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/common.h"
+#include "dali/pipeline/operators/reader/loader/loader.h"
+
+namespace dali {
+
+struct TensorSequence {
+  std::vector<Tensor<CPUBackend>> tensors;
+};
+
+// TODO(klecki) consider using FileLoader as base class
+// TODO(klecki) ?? allow for more high level grouping of sequences:
+//              we should probably make sure that other Loaders read
+//              sequentially, and allow for SequenceLoader to wrap any other
+//              loader, similar for parser and reader
+// TODO(klecki) loader is responsible for handling shuffle_
+class SequenceLoader : public Loader<CPUBackend, TensorSequence> {
+ public:
+  // TODO(klecki) For now sequence is <directory, image list> pair, later it
+  // will be a video file
+  // TODO(klecki) SequenceBucket or other name?
+  using Stream = std::pair<std::string, std::vector<std::string>>;
```

Comment:
Done.
I was thinking about being able to handle the stream from different places but now it is only use in this class. In the case of using it outside, it would probably better to have it defined outside of that class.